### PR TITLE
move sigmoid to core fusion

### PR DESCRIPTION
**Pull request by sasadep**
* support for sigmoid on multiple backend
* sigmoid reference kernels for interpreter backend

### CODE REVIEW

File: src/ngraph/op/sigmoid.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#pragma once
+
+#include "ngraph/op/util/requires_tensor_view_args.hpp"
+#include "ngraph/util.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        class Sigmoid : public util::RequiresTensorViewArgs
```
Comment (aprocter) [inline src/ngraph/op/sigmoid.hpp]: I'm thinking this should inherit from `UnaryElementwiseArithmetic`.

File: src/ngraph/op/sigmoid.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        {
+        public:
+            Sigmoid(std::shared_ptr<Node> input);
+            Shape get_input_shape() const { return m_shape_input; }
```
Comment (aprocter) [inline src/ngraph/op/sigmoid.hpp]: This function is not necessary; we can always use `get_input_shape(0)` (from `Node`)

File: src/ngraph/op/sigmoid.hpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            virtual std::shared_ptr<Node>
+                copy_with_new_args(const NodeVector& new_args) const override;
+            virtual void generate_adjoints(autodiff::Adjoints& adjoints,
+                                           const NodeVector& deltas) override;
+
+        private:
+            Shape m_shape_input;
```
Comment (aprocter) [inline src/ngraph/op/sigmoid.hpp]: This field is not necessary.

**Comment by sasadep** [comment]
@Krovatkin all of points that @aprocter mentioned are addressed

**Comment by sasadep** [comment]
TESTNOW

File: src/ngraph/op/sigmoid.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        throw ngraph_error("Incorrect number of new arguments");
+    }
+
+    return make_shared<Sigmoid>(new_args.at(0));
+}
+
+op::Sigmoid::Sigmoid(shared_ptr<Node> arg)
+    : UnaryElementwiseArithmetic("Sigmoid", {arg})
+{
+    set_value_type_checked(arg->get_element_type(), arg->get_shape());
```
Comment (aprocter) [inline src/ngraph/op/sigmoid.cpp]: It won't hurt anything, but this call is not necessary. (`UnaryElementwise`'s constructor does it for you.)

**Pull request merged**